app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -154,7 +154,7 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
     }
 
     override fun onEngineViewCreated(engineView: EngineView): Disposable? {
-        return serviceLocator?.screenController?.currentActiveScreen?.subscribe {
+        return serviceLocator?.screenController?.currentActiveScreen?.distinctUntilChanged()?.subscribe {
```

Comment:
This is probably going to be a gotcha in other parts of the codebase, too.  I think we can reasonably assume that this observable should always use `distinctUntilChanged`, could you add this operator in `ScreenController` line 37 instead?

Reply:
Yes! Makes sense.

I didn't look into why this was emitting multiple times, which was also strange.